### DISCUSSION TITLE
OrderBy/Limit issue with a ManyToMany

### DISCUSSION THREAD

**Discussion by HugoMrnth** [mikro-orm/mikro-orm#3554] [Q&A] [answered]
Hello

**context** 

I was trying to get a list of clients with their invoices sort DESC by date. Was working, but we decided to link specific users to specific clients ( a company can have several user accounts , but not all these users can access all the clients of the company). We then create a Many to Many bewteen User and Client entities so we could sort the list of clients. 

**Describe the bug**

I have a orderBy being trigered two times during a query with 3 subQuery. 
The orderBy is used to sort the result on one of the subQueries:  eg .orderBy({ '(lastInvoiceDate)': 'DESC' }) . 

Everything was working with one joinAndSelect on one table, a leftJoinAndSelect on another on. But when add a third table with leftJoin (on the many to many relationship, User <=> Client), the orderBy seems to be triggered twice. 

**exemple**

```const offset = paginationFilterQuery.perPage * (paginationFilterQuery.currentPage - 1);
const qb1 = this.clientQueryRepository
.createQueryBuilder('c')
.count('c.id')
.where({ company: currentCompany })
.as('totalRows');

const totalRows = await qb1.count();

const knex = this.clientQueryRepository.getKnex();
let qb3 = this.invoiceQueryRepository
.createQueryBuilder('i')
.select(['createdAt'])
.where({ client: knex.ref('c.id') })
.limit(1)
.as('lastInvoiceDate');

let qb2 = this.clientQueryRepository
.createQueryBuilder('c')
.select(['id', qb3])
.leftJoin('c.users', 'us')
.orderBy({ '(lastInvoiceDate)': 'DESC' })
.limit(paginationFilterQuery.perPage, offset);
```

**here is the error I got : **

select `c`.`id`, (select `i`.`createdAt` from `invoice` as `i` where `i`.`clientId` = `c`.`id` limit 1) as `lastInvoiceDate` from `client` as `c` left join `userHasClient` as `u1` on `c`.`id` = `u1`.`clientId` left join `user` as `us` on `u1`.`userId` = `us`.`id` where `c`.`id` in (select `c`.`id` from (select `c`.`id` from `client` as `c` left join `userHasClient` as `u1` on `c`.`id` = `u1`.`clientId` left join `user` as `us` on `u1`.`userId` = `us`.`id` group by `c`.`id` order by min((lastInvoiceDate)) desc limit 10) as `c`) order by (lastInvoiceDate) desc - Unknown column 'lastInvoiceDate' in 'order clause'

So I think I missed a point somewhere, could you help? 

Thanks a lot and have good day !

**Comment by B4nan** [comment, marked as answer]
Try to remove the parens, this way the ORM does not understand you want that `lastInvoiceDate` subquery, and not just some random string. And register the subquery via `qb.withSubquery()`.

```ts
qb.withSubquery(...).orderBy({ lastInvoiceDate: 'DESC' })
```

**Reply by HugoMrnth** [reply]
Sorry for late reply and thank you for your anwser. Accessing lastInvoiceDate like this isn't possible because it isn't a property of ClientEntity. 
Would you have another idea how we could do this ? Thank you.

**Reply by B4nan** [reply]
Maybe try calling `qb.withSubquery()`, and I am sure youy need to drop the parens, with them present it cant match the key.

But it could be unsupported usecase too. If you could prepare a repro, ideally as a [stackblitz](https://stackblitz.com/edit/mikro-orm-repro), I can take a closer look.

**Reply by HugoMrnth** [reply]
It worked with "qb.withSubquery()", thanks !
